subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java

```diff
@@ -393,7 +386,10 @@ public List<Sign> sign(DomainObjectCollection<Publication> publications) {
     }
 
     private <T extends PublicationArtifact> Sign createSignTaskFor(final PublicationInternal<T> publicationToSign) {
-        final Function<String, Sign> computeSignTask = signTaskName -> {
+        final String signTaskName = determineSignTaskNameForPublication(publicationToSign);
+        if (project.getTasks().getNames().contains(signTaskName)) {
+            return project.getTasks().named(signTaskName, Sign.class).get();
+        } else {
```

Comment:
What do you think of removing else branch to reduce indentation level?
If branch finishes with return, so there's no need in else.